src/main/resources/META-INF/services/sun.net.spi.nameservice.NameServiceDescriptor

```diff
@@ -0,0 +1 @@
+org.broadinstitute.dsde.workbench.leonardo.dns.JupyterNameServiceDescriptor
```

Comment:
wat

Reply:
okay, so _maybe_ i'm following: this is a classpath of a thing that someone (docker? can i see the compose? it's not in the linked PR in fc-dev) instantiates to make an in-memory DNS server that resolves fake jupyter.leo addresses to real IPs?

But then `getTargetHost` in `ProxyService` queries the `clusterDnsCache` actor directly, so it's not a magic host-level thing. I don't understand after all.

Reply:
Partly right, this instantiates an in-memory fake DNS server that resolves fake jupyter.firecloud.org addresses to real IPs.

But it has nothing to do with docker, the Java container instantiates it w/ reflection based on the classname in this "special" file.